sdk/sdk-api/src/main/java/mil/emp3/api/utils/ColorUtils.java

```diff
@@ -41,4 +42,26 @@ public static String colorToString(IGeoColor oColor) {
         
         return sStr;
     }
+
+    /**
+     * Returns the IGeoColor of the given hex string
+     * @param hexColorValue - Hex color in String form. Assuming the form of #RRGGBB
+     * @return - IGeoColor object representing the given string.
+     */
+    public static IGeoColor stringToColor(final String hexColorValue) {
+        final int radix = 16;
+        try {
+            int red = Integer.valueOf(hexColorValue.substring(1, 3), radix);
+            int green = Integer.valueOf(hexColorValue.substring(3, 5), radix);
+            int blue = Integer.valueOf(hexColorValue.substring(5, 7), radix);
+            final IGeoColor geoColor = new GeoColor();
+            geoColor.setRed(red);
+            geoColor.setGreen(green);
+            geoColor.setBlue(blue);
+            return geoColor;
+        }
+        catch (Exception exception) {
+            throw new RuntimeException("Color string was not formatted properly. Proper format is #RRGGBB.");
```

Comment:
A runtime exception is unchecked and *fatal*, hence they should be avoided at all costs.  `stringToColor` needs to be able to recover gracefully from a situation in which the string didnt parse out a good color.  As Jen mentioned, catching the right exception is phase 1, phase 2 should not be throwing an unchecked exception but instead:

1. log the fact that an unsupported string color value was requested

Then either:

2a. return a default or fallback color
2b. throw a better exception that details the problem and handle it in the calling class

We have been a bit lax in other projects at enforcement of good exception practices so I wanted to make sure I outlined the issue clearly for everyone here.  The blame for that rests with me. :disappointed:

Reply:
Fixed. Returning black as default which is what EMP was doing previously when you gave it a bad color.